File: app/src/main/java/com/example/qiuchenly/comicparse/UI/Main/MainPresenter.java
package com.example.qiuchenly.comicparse.UI.Main;

import com.example.qiuchenly.comicparse.Bean.HotComicStrut;
import com.example.qiuchenly.comicparse.Simple.BasePresenterImp;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

/**
 * 主页P层实现类
 * 作者 秋城落叶
 * 2018年7月30号
 */
public class MainPresenter extends BasePresenterImp<MainContract.View, MainModel> implements MainContract.Presenter {

    public MainPresenter(MainContract.View mView) {
        super(mView);
    }

    @Override
    public void getHotComic() {
        SuperModel.getHotsComic(new MainContract.GetHotComic() {
            @Override
            public void onSuccessGetHot(@NotNull ArrayList<HotComicStrut> arr) {
                if (isShow()) SuperView.getHotComicList(arr);
            }

            @Override
            public void onFailed(@NotNull String reasonStr) {
                if (isShow()) SuperView.ShowErrorMsg(reasonStr);
            }
        });
    }

    @Override
    public MainModel createModel() {
        return new MainModel();
    }
}

File: ComicX/src/main/java/com/qiuchenly/comicparse/MVP/Presenter/MainPresenter.kt
package com.qiuchenly.comicparse.MVP.Presenter

import com.qiuchenly.comicparse.Modules.MainActivity.Fragments.TuiJian.Beans.HotComicStrut
import com.qiuchenly.comicparse.MVP.Contract.MainContract
import com.qiuchenly.comicparse.MVP.Model.Fragment_MainModel
import java.util.*

/**
 * 主页P层实现类
 * 作者 秋城落叶
 * 2018年7月30号
 */
class MainPresenter(private var mView: MainContract.View) {
    var model = Fragment_MainModel()
    fun getHotComic() {
        model.getHotsComic(object : MainContract.GetHotComic {
            override fun onSuccessGetHot(arr: ArrayList<HotComicStrut>) {
                if (mView != null) mView!!.getHotComicList(arr)
            }

            override fun onFailed(reasonStr: String) {
                if (mView != null) mView!!.ShowErrorMsg(reasonStr)
            }
        })
    }
}